### ISSUE TITLE
hmake test gives Credential Chain error

### ISSUE THREAD

**Issue by aaron-sua** [vmware/vmware-go-kcl#3]
[test]              	worker_test.go:77: Errorin Publish. NoCredentialProviders: no valid providers in chain. Deprecated.
[test]              			For verbose messaging see aws.Config.CredentialsChainVerboseErrors

aws ec2 describe-instances works without issue however from the same system

**Comment by taojwmware** [comment]
You might want to do the following in the test:

func TestWorker(t *testing.T) {
	os.Setenv("AWS_ACCESS_KEY_ID", "your aws access key id")
	os.Setenv("AWS_SECRET_ACCESS_KEY", "your aws secret access key")
	defer os.Unsetenv("AWS_ACCESS_KEY_ID")
	defer os.Unsetenv("AWS_SECRET_ACCESS_KEY")

	kclConfig := cfg.NewKinesisClientLibConfig("appName", streamName, regionName, workerID).
		WithInitialPositionInStream(cfg.LATEST).
		WithMaxRecords(40).
		WithMaxLeasesForWorker(1).
		WithShardSyncIntervalMillis(5000)

hmake test runs inside a docker contain and it may not be able to access the credentials under ~/.aws

**Comment by aaron-sua** [comment]
actually I modified the hmake file to pass the env vars which worked.  But leaving hard coded credentials isn't great.  I ended up using the volumes declaration.

    volumes:
        - '~/.aws:/root/.aws:ro'

However ~ doesn't always expand on the various environments this needs to run in.  
Unfortunately hmake doesn't have a good way to pass variables at runtime it seems.
So I've used a .hmakerc file with the settings override.

    volumes:
        - 'Users/myLocalUser/.aws:/root/.aws:ro'

HyperMake is interesting but most of the time I use Ansible for this sort of thing as it provides a lot more flexibility across environments and keeps everything unified.
